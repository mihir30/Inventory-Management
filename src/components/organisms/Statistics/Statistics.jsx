import React, { useEffect, useState } from 'react'
import ChartSection from '../../molecules/ChartSection/ChartSection'
import FilterSectionStatistics from '../../molecules/FilterSectionStatistics/FilterSectionStatistics'
import NavigationBar from '../../molecules/NavigationBar/NavigationBar'
import { getData } from '../../utils/common.utils'

const Statistics = () => {
    const [unitsList, setUnitsList] = useState([]);
    const url = window.location.href;
    useEffect(() => {
        const fetchData = async () => {
            try {
                const units = await getData('http://localhost:8000/units');
                setUnitsList(units);

            } catch (error) {
                console.log("error", error.message);
            }
        };
        fetchData();

    }, [url]);

    const [fromDate, setItems] = useState([]);
    const [type, setType] = useState("Computer")
    useEffect(() => {
        const fromDateList = JSON.parse(localStorage.getItem('dateList'));
        if (fromDateList) {
            setItems(fromDateList);
        }

        const typevalue = JSON.parse(localStorage.getItem('type'));
        if (typevalue) {
            setType(typevalue);
        }


    }, [fromDate, type]);

    const filterUnitsSection = () => {

        const results = unitsList.filter(element => {
            return fromDate.includes(element.fromDate) && element.type === type;
        });

        return results;

    }

    const results = filterUnitsSection();

    return (
        <div><NavigationBar></NavigationBar>
            <FilterSectionStatistics></FilterSectionStatistics>
            <ChartSection statusResults={results}></ChartSection>
        </div>

    )
}

export default Statistics