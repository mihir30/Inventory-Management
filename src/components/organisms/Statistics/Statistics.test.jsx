import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ReactDom from "react-dom";
import Statistics from './Statistics';

describe("Statistics Component", () => {
    
    const filterUnitsSection = jest.fn(() => {
        return [];
    })

    const results = filterUnitsSection();

    it('renders the component properly', () => {
        const divEl = document.createElement("div");
        ReactDom.render(<BrowserRouter><Statistics></Statistics></BrowserRouter>, divEl); 
    });

    it('should return results correctly', () => {
        expect(results).toEqual([]);
    })

    it('should have the right statistics component snapshot', () => {
        const tree = renderer.create().toJSON();
        expect(tree).toMatchSnapshot();
    })

});