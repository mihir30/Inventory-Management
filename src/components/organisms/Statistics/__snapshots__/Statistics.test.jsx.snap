// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statistics Component should have the right statistics component snapshot 1`] = `null`;
